textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcher.java

```diff
@@ -34,22 +51,45 @@ public void open() throws DataFlowException {
 
     @Override
     public ITuple getNextTuple() throws DataFlowException {
-
-        try {
+		try {
             ITuple sourceTuple = sourceOperator.getNextTuple();
             if(sourceTuple == null){
                 return null;
             }
-            if (predicate.satisfy(sourceTuple)) {
-                return sourceTuple;
-            } else {
-                return getNextTuple();
+            
+            RegexPredicate regexPredicate = (RegexPredicate)predicate; 
+            
+            spans = regexPredicate.computeMatches(sourceTuple);
+            
+            if (spans != null && spans.size() != 0) { // a list of matches found
+            	if (schema == null) {
+            		schema = sourceTuple.getSchema();
+            	}
+            	if (spanSchema == null) {
+            		spanSchema = Utils.createSpanSchema(schema);
+            	}
+            	fields = sourceTuple.getFields();
+            	return getSpanTuple(spans);
+            } else { // no match found
+            	return getNextTuple();
             }
         } catch (Exception e) {
             e.printStackTrace();
             throw new DataFlowException(e.getMessage(), e);
-        }
-
+        }        
+    }
+    
+    private ITuple getSpanTuple(List<Span> spans) {
+    	List<IField> fieldListDuplicate = new ArrayList<>(fields);
```

Comment:
@sandeepreddy602 : please check if this function is consistent with the way the dictionary matcher constructs a span tuple, and whether some code can be refactored.